### ISSUE TITLE
tt_lib softmax_in_place fails for some input values [Wormhole]

### ISSUE THREAD

**Issue by nemanjagrujic** [tenstorrent-metal/tt-metal#3601]
**tt_lib.operations.primary.softmax_in_place** operation breaks with low PCC error in all test cases.

**To Reproduce**
1. Checkout `main` branch
2. Run unit test `test_eltwise_softmax_in_place.py ` using the command:
`pytest tests/tt_eager/python_api_testing/non_working_unit_tests/grayskull/test_eltwise_softmax_in_place.py`
[Same unit test as for grayskull]

**Expected behavior**
There are three test cases presented in the unit test `test_eltwise_softmax_in_place.py`. All of them are expected to fail with low PCC error like this:
`Max ATOL Delta: 1.0, Max RTOL Delta: nan, PCC: 0.041026371396878716, PCC check failed`

**Getting Additional info for the operation under test and its behavior**
To get additional information about different combinations of input shapes, types, layouts and memory configs for which this operation was tested, you can run locally **sweeps for softmax_in_place** and check the results. To do this you should:
1. Follow the `Getting Started` page to set up the repo, environment variables and `python-env`
2. Activate `source build/python_env/bin/activate`
3. Run sweeps by using next command:
`python tests/tt_eager/python_api_testing/sweep_tests/run_pytorch_test.py -i tests/tt_eager/python_api_testing/sweep_tests/test_configs/ci_sweep_tests/broken_wormhole/pytorch_eltwise_softmax_in_place_test.yaml -o ./result-sweeps`
4. After the run is completed all test sweeps results should be available inside specified output directory (in this case ./result-sweeps). There you will find `eltwise_softmax_in_place_sweep.csv` which holds all executed tests, among which you can also find the ones that failed and the one which passed,  and were recreated by the unit test, and which you can locate by searching the unique `data_seed` field.

**Comment by yugaoTT** [comment]
the shape passed from my unit test, perhaps unit test error

**Comment by nemanjagrujic** [comment]
@yugaoTT I checked your unit test (is it tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax.py).

The difference what makes it fall is range of input values. If you change line 29 to:

input_tensor = torch.Tensor(size=input_shape).uniform_(-100, 100)

you will make unit test fail.

Is there any limit on range of input values for this op? If yes is it documented?

**Comment by yugaoTT** [comment]
similar to scale, the input is too large. we don't have it documented, not clear at which point it will fail. But in the actual model, we won't expect values this large

**Comment by nemanjagrujic** [comment]
Closing this issue since new ticket is created to improve numerical stability of softmax:  https://github.com/tenstorrent-metal/tt-metal/issues/7148
